src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -332,20 +332,23 @@ private void ILVerify(Verification verification)
             // Main module is the first one
             var mainModuleReader = resolver.Resolve(_allModuleData[0].SimpleName);
 
-            var (succeeded, errorMessage) = verify(verifier, mscorlibModule.SimpleName, mainModuleReader);
+            var (actualSuccess, actualMessage) = verify(verifier, mscorlibModule.SimpleName, mainModuleReader);
+            var expectedSuccess = !verification.Status.HasFlag(VerificationStatus.FailsILVerify);
 
-            switch (succeeded, (verification & Verification.FailsILVerify) == 0)
+            if (actualSuccess != expectedSuccess)
             {
-                case (true, true):
-                    return;
-                case (true, false):
-                    throw new Exception("IL Verify succeeded unexpectedly");
-                case (false, false):
-                    return;
-                case (false, true):
-                    throw new Exception("IL Verify failed unexpectedly: \r\n" + errorMessage);
+                throw new Exception(expectedSuccess ?
+                    "IL Verify failed unexpectedly: \r\n" + actualMessage :
```

Comment:
I would prefer:
```cs
$"""
IL Verify failed unexpectedly:
{actualMessage}
"""
```

Reply:
Verbatim string formats really badly in the ternary expression, but I'm gonna use interpolation instead.